src/bp/ui-studio/src/web/components/Layout/index.tsx

```diff
@@ -256,6 +276,7 @@ const Layout: FC<ILayoutProps & StateProps> = props => {
 }
 
 const mapStateToProps = (state: RootReducer) => ({
+  currentSession: state.nlu.trainSession,
```

Comment:
Adding the train session to the main layout class will trigger a re-render everytime the train session is updated (which is each second right now)... Can you move this logic to a child component instead ?